docs/image_previews.md

```diff
@@ -0,0 +1,231 @@
+
+## Joshuto Wrapper
+First, we need a wrapper script for Joshuto.
+Joshuto is not started directly anymore but through that wrapper script.
+Place this script in a directory which is in your `$PATH`.
+You may name the script `j` or `jo` or whatever you like to type to start Joshuto.
+You may also name it `joshuto`, just take care that it
+must come before the actual `joshuto` binary in you `$PATH` in that case.
+
+```bash
+#!/usr/bin/env bash
+
+if [ -n "$DISPLAY" ] && command -v ueberzug > /dev/null; then
+    export joshuto_wrap_id="$$"
+    export joshuto_wrap_tmp="$(mktemp -d -t joshuto-wrap-$joshuto_wrap_id-XXXXXX)"
+    export joshuto_wrap_ueber_fifo="$joshuto_wrap_tmp/fifo"
+    export joshuto_wrap_pid_file="$joshuto_wrap_tmp/pid"
+    export joshuto_wrap_preview_meta="$joshuto_wrap_tmp/preview-meta"
+    export joshuto_wrap_ueber_identifier="preview"
+
+    function start_ueberzug {
+        mkfifo "${joshuto_wrap_ueber_fifo}"
+        tail --follow "$joshuto_wrap_ueber_fifo" | ueberzug layer  --parser bash &
+        echo "$!" > "$joshuto_wrap_pid_file"
+        mkdir -p "$joshuto_wrap_preview_meta"
+    }
+
+    function stop_ueberzug {
+        ueberzug_pid=`cat "$joshuto_wrap_pid_file"`
+        kill "$ueberzug_pid"
+        rm -rf "$joshuto_wrap_tmp"
+    }
+
+    function show_image {
+        >"${joshuto_wrap_ueber_fifo}" declare -A -p cmd=( \
+                [action]=add [identifier]="${joshuto_wrap_ueber_identifier}" \
+                [x]="${2}" [y]="${3}" \
+                [width]="${4}" [height]="${5}" \
+                [path]="${1}")
+    }
+
+    function remove_image {
+        >"${joshuto_wrap_ueber_fifo}" declare -A -p cmd=( \
+            [action]=remove [identifier]="${joshuto_wrap_ueber_identifier}")
+    }
+
+    function get_preview_meta_file {
+        echo "$joshuto_wrap_preview_meta/$(echo "$1" | md5sum | sed 's/ //g')"
+    }
+
+    export -f get_preview_meta_file
+    export -f show_image
+    export -f remove_image
+ 
+    trap stop_ueberzug EXIT QUIT INT TERM
+    start_ueberzug
+    echo "ueberzug started"
+fi
+
+joshuto "$@"
+exit $?
+```
+
+The script must be _executable_!
+
+This script starts an “ueberzug server” and then Joshuto itself.
+It takes care that ueberzug is stopped when
+`joshuto` terminates.
+Each Joshuto instance will have its own instance of an “ueberzug server”.
+The script also provides some functions
+and variables which can be used in sub-processes.
+
+## Configuring Hook Scripts
+When started with the wrapper script, Joshuto's sub-processes can show and remove a
+preview image with Überzug now. Joshuto offers two hooks which will be used for that.
+
+In your `~/.config/joshuto/joshuto.toml`, configure a script for each of these hooks: 
+
+```toml
+[preview]
+...
+preview_shown_hook_script = "~/.config/joshuto/on_preview_shown"
+preview_removed_hook_script = "~/.config/joshuto/on_preview_removed"
+```
```

Comment:
what do you think about naming them
```bash
preview_on_show_script
preview_on_remove_script
```

Reply:
Hm, not my native language. For my ears it sounds like the scripts would provide a preview when some "show" or "remove" event happens. Not like the scipt is called _as a result_ of Joshuto's preview being shown or removed.

Maybe
```
on_preview_show_script
on_preview_remove_script
```
?